packages/rest/src/lib/RequestManager.ts

```diff
@@ -28,7 +30,7 @@ export interface RawFile {
 	/**
 	 * The actual data for the file
 	 */
-	rawBuffer: Buffer;
+	fileData: string | number | boolean | Buffer | ReadStream | ServerResponse;
```

Comment:
All of these values other than string aren't [spec compliant](https://developer.mozilla.org/en-US/docs/Web/API/FormData/append) and will make using a different library harder.

Reply:
That's not actually true, the only non-spec compliant are the streams, which based on vlads comment above may be unnecessary anyways. The spec calls for .toString() to be callled on anything that is not a Blob or USVstring

Reply:
The spec doesn't have value converted to a USV string though? https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#create-an-entry

Reply:
Interesting that it is not in the bare bones html spec, however both the [web api](https://developer.mozilla.org/en-US/docs/Web/API/FormData/append) and [undici](https://github.com/nodejs/undici/blob/main/lib/fetch/formdata.js#L37) convert any non string / blob into a string

Regardless, when we do switch to undici, the second argument can *only* be of type `Blob` when providing a third argument, so we are going to need to translate anyways since Discord now essentially requires the third parameter

Reply:
From experience the spec will often leave basic things out of the spec and have the reader infer what their implementation should do. In this case, the param should theoretically be `any` other than Symbols and objects with a Symbol.toStringTag that throws an error I guess.

However, since it's written in ts, the values will need to be converted to string/Blob before being able to append them anyways, once a switch is made.